README.md

```diff
@@ -250,6 +251,119 @@ class UsersController < ApplicationController
 end
 ```
 
+### Custom input errors
+
+Use a `Hash` with `is:` and `message:` keys to prepare custom
+error messages on inputs. For example:
+
+```rb
+class UpdateAdminUser < Actor
+  input :user,
+        must: {
+          be_an_admin: {
+            is: -> user { user.admin? },
+            message: "The user is not an administrator"
+          }
+        }
+
+  # ...
+end
+```
+
+You can also use incoming arguments when shaping your error text:
+
+```rb
+class UpdateUser < Actor
+  input :user,
+        allow_nil: {
+          is: false,
+          message: (lambda do |_origin, input_key, _service_name|
```

Comment:
Instead of relying on positional arguments, I think it would be simpler to use named arguments for messages.

For example, here, it would simplify this line to:

```suggestion
          message: (lambda do |input_key:|
```

Reply:
@sunny I tried using named keywords but as a result I started getting this error when the required keys are missing:

```ruby
message: (lambda do |input_key:|
```

> ArgumentError: unknown keywords: :origin, :service_name

Maybe I'm confused in the logic at the moment 😅

Reply:
Oh, sorry 🙈

Since they are lambdas, users should use the following syntax instead to account for the other named arguments:

```rb
message: (lambda do |input_key:, **|
```

Or, perhaps better, to use procs instead which don’t require defining all arguments:

```rb
message: (proc do |input_key:|
```

Reply:
Replacing it with `proc` did not solve the problem. But adding `**` in both cases helped.

Reply:
@sunny Example: https://github.com/sunny/actor/pull/82/commits/48cec31044e9de2098b0741a263c3fb5e10ffbe9

Reply:
The problem is not that some attributes are not passed and their obligation must be ignored through `proc`. All attributes are passed. The problem is that inside the lambda there are attributes that the lambda did not expect. This is true for `proc` as well.

Reply:
The problem is not that some attributes are not passed and their obligation must be ignored through `proc`. All attributes are passed. The problem is that inside the lambda there are attributes that the lambda did not expect. This is true for `proc` as well.

This only applies to named arguments.

Reply:
Oh, you are absolutely right, named arguments are not ignored in procs like they are with positional arguments 🤦🏻

This introduces an issue for the future as we don’t want to have breaking changes every time we introduce new arguments.

We could:
- encourage users to add `**` to the end of all theirs calls even when they have all arguments (which can be a bit tedious).
- encourage getting errors as a single hash, instead:

```rb
message: -> _ { "Cannot be blank }
message: -> error { "#{error[:input_key]} cannot be blank" }
```

I think this syntax would be more future-friendly.